alacritty_terminal/src/term/mod.rs

```diff
@@ -641,6 +641,9 @@ pub struct SizeInfo {
     /// Number of lines in the viewport.
     screen_lines: Line,
 
+    /// Number of total visible lines, including screen_lines and reserved lines.
+    visible_lines: Line,
```

Comment:
Why would the terminal need to know this, that makes little sense to me.

All the terminal should have to know, is how big the **terminal** is. Everything else has no business being stored in the terminal.

Reply:
I was unaware of this scope difference. I'll revert this and try find another way of tracking the total count of lines on screen.

Reply:
>I was unaware of this scope difference. I'll revert this and try find another way of tracking the total count of lines on screen.

You can compute it from padding/height, and cell height somewhere in alacritty crate, just not in alacritty_terminal, since size info is purely about terminal.